tests/js/spec/utils/useProjects.spec.tsx

```diff
@@ -0,0 +1,94 @@
+import {act, renderHook} from '@testing-library/react-hooks';
+
+import OrganizationStore from 'app/stores/organizationStore';
+import ProjectsStore from 'app/stores/projectsStore';
+import useProjects from 'app/utils/useProjects';
+
+describe('useProjects', function () {
+  const org = TestStubs.Organization();
+
+  const mockProjects = [TestStubs.Project()];
+
+  it('provides projects from the team store', function () {
+    act(() => void ProjectsStore.loadInitialData(mockProjects));
+
+    const {result} = renderHook(() => useProjects());
+    const {projects} = result.current;
+
+    expect(projects).toEqual(mockProjects);
+  });
+
+  it('loads more projects when using onSearch', async function () {
+    act(() => void ProjectsStore.loadInitialData(mockProjects));
+    act(() => void OrganizationStore.onUpdate(org, {replace: true}));
+
+    const newProject3 = TestStubs.Project({id: '3', slug: 'test-project3'});
+    const newProject4 = TestStubs.Project({id: '4', slug: 'test-project4'});
+
+    const mockRequest = MockApiClient.addMockResponse({
+      url: `/organizations/${org.slug}/projects/`,
+      method: 'GET',
+      body: [newProject3, newProject4],
+    });
+
+    const {result, waitFor} = renderHook(() => useProjects());
+    const {onSearch} = result.current;
+
+    // Works with append
+    const onSearchPromise = act(() => onSearch('test'));
+
+    expect(result.current.fetching).toBe(true);
+    await onSearchPromise;
```

Comment:
```suggestion
    // Works with append
    await act(() => onSearch('test'));
```

Reply:
I can't move this around like this in order to test the `fetching` state